### PR TITLE
Fixes issue with root iframe load event timing.

### LINKED ISSUE #83: Root test suites can bail via “Timed out loading …” when iframe is deleted before load events fire.
## Context

This issue only manifests when you visit the browser manually (not via something like Puppeteer). It sows confusion because it makes it look like the _root_ test failed even though it is actually because a _child test hung_. As a developer, you are basically served a red herring in this edge case. The best way to explain is to drop a simple example and show images.

## Proposal

Simply catch the edge case where the root test exited _so fast_ that we hit a race condition in the browser where the load event on the iframe was never heard (because the child suite was ready, but then had no actual tests to run). We can do this my saving an additional piece of state when the child suite declares that it’s `ready`.

## Example

```html
<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="content-security-policy" content="default-src 'self';">
  </head>
  <body>
    <h3>demo stale-race guard</h3>
    <!--
      Open in a real browser (not Puppeteer). This root has no tests of
      its own — it only registers a slow sibling test. The sequence:
        1. Root boots, creates self-iframe for this same URL.
        2. Self-iframe loads (cached), publishes `ready`, registers the
           sibling, and reports completion before its own iframe.load
           dispatches on the parent.
        3. Parent kicks off the sibling, which removes the self-iframe.
           Its pending load event is now discarded.
        4. ~30s later the self-iframe's load-phase timer fires. Without
           the stale-race guard the run bails with "Timed out loading".
           With the guard, the stale timer is ignored and the demo
           completes cleanly ~35s in.
    -->
    <script type="module" src="./index.js"></script>
  </body>
</html>
```

```js
import { test } from '../../x-test.js';

test('./keepalive.html');
```

**Before:**

(note how it used to suggest that the issue was from loading the _root test_ — which is _not true_)

Image

**After:**

Image

### PR DESCRIPTION
**Pull request by theengineear**
The crux of this issue is fixing mis-reporting for some edge cases that commonly happen when running tests.

The symptom is that the reporter can mis-diagnose the issue as a problem with the root test (thinking that the root test timed out), when the actual failure (or simply slow test) is in a child sub-test iframe.

The fix it to catch the edge case where the timeout fires on the root, but we can detect that the tests _did_ actually run.

Closes #83.

### CODE REVIEW

File: demo/stale-race/index.html [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        1. Root boots, creates self-iframe for this same URL.
+        2. Self-iframe loads (cached), publishes `ready`, registers the
+           sibling, and reports completion before its own iframe.load
+           dispatches on the parent.
+        3. Parent kicks off the sibling, which removes the self-iframe.
+           Its pending load event is now discarded.
+        4. ~30s later the self-iframe's load-phase timer fires. Without
+           the stale-race guard the run bails with "Timed out loading".
+           With the guard, the stale timer is ignored and the demo
+           completes cleanly ~35s in.
```
Comment (theengineear) [inline demo/stale-race/index.html:21]: ☝️ — This is the complete repro. Yes, it was hard to simply nail this down. I have intermittently hit this issue while testing _many times_ because I got into a state where some child test was simply running slow or hanging. The reporter would mislead me into thinking the issue was actually _something else_, which was frustrating. Interestingly, this isn’t reproducible via Puppeteer, which made this situation even more cryptic.

_Both_ environments should either correctly succeed or correctly report the issue now!
